tag-master goal unable to push to BitBucket

When merging branches into master with the tag-master goal enabled I get:

```
[INFO] Executing: /bin/sh -c cd /var/lib/jenkins/jobs/myProject/workspace && git push http:********@git.myServer.com/myProject.git refs/tags/0.0.37
[INFO] Working directory: /var/lib/jenkins/jobs/myProject/workspace
[ERROR] Provider message:
[ERROR] The git-push command failed.
[ERROR] Command output:
[ERROR] fatal: could not read Password for 'http://jenkins@git.myServer.com': No such device or address
```

In addition to the gitflow-helper-maven-plugin my pom.xml I also has:

```
<project.scm.id>myServerId</project.scm.id>
...
<scm>
    <connection>scm:git:http://jenkins@git.myServer.com/scm/myProject.git</connection>
    <developerConnection>scm:git:http://jenkins@git.myServer.com/scm/myProject.git</developerConnection>
    <tag>HEAD</tag>
</scm>
```

and in the settings.xml of the docker container running Jenkins I have:

```
<servers>
    <server>
        <id>myServerId</id>
        <username>jenkins</username>
        <password>myPassword</password>
    </server>
    ...
</servers>
```

I notice that when I use the maven-release-plugin my git push looks more like:
`[INFO] Executing: /bin/sh -c cd /var/lib/jenkins/jobs/myProject/workspace && git push http://jenkins:********@git.myServer.com/scm/myProject.git refs/tags/0.0.37`
that is to say, it has "//jenkins:" after the username. I'm guessing that somehow my username and password from the settings.xml is not being added to the push request when using gitflow-helper-maven-plugin.

I'm having the same trouble when running the build on Atlassian Bamboo. 

It looks like the provided Git URL is never used although it is being set as the developerConnectionUrl. The reason is that the connectionType is not being set as "developerConnection" but has the default value of "connection".

Here is some output from the Maven run on the Bamboo agent:

```
[DEBUG] Configuring mojo 'org.apache.maven.plugins:maven-scm-plugin:1.9.4:tag' with basic configurator -->
[DEBUG]   (f) tag = V1.1
[DEBUG]   (f) developerConnectionUrl = scm:git:scm:git:ssh://git@xxx/yyy/zzz.git
[DEBUG]   (f) addTimestamp = false
[DEBUG]   (f) basedir = /var/atlassian/bamboo-agent-home/xml-data/build-dir/xxx
[DEBUG]   (f) connectionType = connection
[DEBUG]   (s) connectionUrl = scm:git:https://xxx/yyy/zzz.git
[DEBUG]   (f) pushChanges = true
[DEBUG]   (f) remoteTagging = true
[DEBUG]   (f) settings = org.apache.maven.execution.SettingsAdapter@64a1923a
[DEBUG]   (f) timestampFormat = yyyyMMddHHmmss
[DEBUG]   (f) timestampPosition = end
[DEBUG]   (f) timestampPrefix = -
[DEBUG] -- end configuration --
[INFO] Final Tag Name: 'V1.1'
[INFO] Executing: /bin/sh -c cd /var/atlassian/bamboo-agent-home/xml-data/build-dir/xxx && git tag -F /tmp/xxx/maven-scm-493794421.commit V1.1
[INFO] Working directory: /var/atlassian/bamboo-agent-home/xml-data/build-dir/xxx
[INFO] Executing: /bin/sh -c cd /var/atlassian/bamboo-agent-home/xml-data/build-dir/xxx && git push https://xxx/yyy/zzz.git refs/tags/V1.1
[INFO] Working directory: /var/atlassian/bamboo-agent-home/xml-data/build-dir/xxx
[ERROR] Provider message:
[ERROR] The git-push command failed.
[ERROR] Command output:
[ERROR] fatal: could not read Username for 'https://xxx': No such device or address
``` 

I think the correct fix would be to add element `connection` with value `developerConnection` to the SCM plugin configuration.

--- a/src/main/java/com/e_gineering/maven/gitflowhelper/TagMasterMojo.java
+++ b/src/main/java/com/e_gineering/maven/gitflowhelper/TagMasterMojo.java
@@ -47,6 +47,9 @@ public class TagMasterMojo extends AbstractGitflowBranchMojo {
                 gitURLExpression = "${env.GIT_URL}";
             }
             String gitURL = resolveExpression(gitURLExpression);
+            if (!gitURL.startsWith("scm:git:")) {
+                gitURL = "scm:git:" + gitURL;
+            }
             getLog().debug("gitURLExpression: '" + gitURLExpression + "' resolved to: '" + gitURL + "'");
             ExpansionBuffer eb = new ExpansionBuffer(gitURL);
             if (!eb.hasMoreLegalPlaceholders()) {
@@ -63,7 +66,8 @@ public class TagMasterMojo extends AbstractGitflowBranchMojo {
                         goal("tag"),
                         configuration(
                                 element(name("tag"), tag),
-                                element(name("developerConnectionUrl"), "scm:git:" + gitURL)
+                                element(name("connectionType"), "developerConnection"),
+                                element(name("developerConnectionUrl"), gitURL)
                         ),
                         executionEnvironment(
                                 project,

